pymc3/tests/test_distributions.py

```diff
@@ -2669,74 +2642,140 @@ def ref_pdf(value):
                 self.check_logp(TestedInterpolated, R, {}, ref_pdf)
 
 
-@pytest.mark.xfail(reason="Bound not refactored yet")
-def test_bound():
-    np.random.seed(42)
-    UnboundNormal = Bound(Normal)
-    dist = UnboundNormal.dist(mu=0, sigma=1)
-    # assert dist.transform is None
-    assert isinstance(dist.random(), np.ndarray)
-
-    LowerNormal = Bound(Normal, lower=1)
-    dist = LowerNormal.dist(mu=0, sigma=1)
-    assert logp(dist, 0).eval() == -np.inf
-    # assert dist.transform is not None
-    assert np.all(dist.random() > 1)
-
-    UpperNormal = Bound(Normal, upper=-1)
-    dist = UpperNormal.dist(mu=0, sigma=1)
-    assert logp(dist, -0.5).eval() == -np.inf
-    # assert dist.transform is not None
-    assert np.all(dist.random() < -1)
-
-    ArrayNormal = Bound(Normal, lower=[1, 2], upper=[2, 3])
-    dist = ArrayNormal.dist(mu=0, sigma=1, size=2)
-    assert_equal(logp(dist, [0.5, 3.5]).eval(), -np.array([np.inf, np.inf]))
-    # assert dist.transform is not None
-    with pytest.raises(ValueError) as err:
-        dist.random()
-    err.match("Drawing samples from distributions with array-valued")
-
-    with Model():
-        a = ArrayNormal("c", size=2)
-        assert_equal(a.tag.test_value, np.array([1.5, 2.5]))
-
-    lower = at.vector("lower")
-    lower.tag.test_value = np.array([1, 2]).astype(aesara.config.floatX)
-    upper = 3
-    ArrayNormal = Bound(Normal, lower=lower, upper=upper)
-    dist = ArrayNormal.dist(mu=0, sigma=1, size=2)
-    logp_dist = logp(dist, [0.5, 3.5]).eval({lower: lower.tag.test_value})
-    assert_equal(logp_dist, -np.array([np.inf, np.inf]))
-    assert dist.transform is not None
+class TestBound:
+    """Tests for pm.Bound distribution"""
 
-    with Model():
-        a = ArrayNormal("c", size=2)
-        assert_equal(a.tag.test_value, np.array([2, 2.5]))
```

Comment:
We need tests that cover the case where lower / upper is a vector, which were covered in these `ArrayNormal` checks